cql/ManageRareAbnormality.cql

```diff
@@ -627,6 +630,70 @@ define HistologyInterpretedAsCin3:
         aC in Dash."Histologic CIN3"
   )
 
+define AnyHistologyInterpretedAsCin3:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC in Dash."Histologic CIN3"
+  )
+
+define AnyHistologyInterpretedAsCin2:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostSevereHistologyInterpretedAsCin2InPast3Years:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostRecentCin2BiopsyInPast3Years:
+  Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 3 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."CIN2"
+    )
+  )
+
+define MostRecentCin2BiopsyInPast2Years:
+  Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 2 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."CIN2"
+    )
+  )
+
+define MostRecentHsilBiopsyInPast3Years:
+    Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 3 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."HSIL, Unspecified"
+    )
+  )
+
+ define MostRecentHsilBiopsyInPast2Years:
+    Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 2 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."HSIL, Unspecified"
+    )
+  ) 
```

Comment:
Both of these expressions (`MostRecentHsilBiopsyInPast3Years` and  `MostRecentHsilBiopsyInPast2Years`) should be named "Any" instead of "MostRecent" to avoid confusion, since these definitions do not look for the most recent biopsy, but rather any biopsy that has a value of "HSIL, Unspecified"

Reply:
These are iterating over the SortedBiopsyReports list, which goes from newest to oldest. Coalesce returns the first match, which is the 'most recent' of the matches.

Reply:
They are iterating over a list of "HSIL, Unspecified" results, not a list of all biopsy results. This is because of the `where` clause you have here. `Coalesce` is being called after you do the filtering with the `where` clause. I can't quite tell by your comment whether that was intentional or not, but this does not seem to align with the use of "Most Recent" in other parts of our CQL logic, where we take take the most recent test from the unfiltered list of SortedBiopsyReports. See https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/118#discussion_r1372354243, as this is related.

Reply:
Wow I add some redundant comments, sorry about that. This was clarified by the use of "LastKnownX" to distinguish it from other areas where we use "MostRecentX"